bundles/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java

```diff
@@ -549,18 +528,38 @@ public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
      */
     private void setDeviceConfiguration(Device device) {
         try {
+            boolean config_changed = false;
             logger.debug("MAX! {} {} configuration update", device.getType(), device.getSerialNumber());
             Configuration configuration = editConfiguration();
-            configuration.put(PROPERTY_ROOMNAME, device.getRoomName());
-            configuration.put(PROPERTY_ROOMID, new BigDecimal(device.getRoomId()));
-            configuration.put(PROPERTY_DEVICENAME, device.getName());
-            configuration.put(PROPERTY_RFADDRESS, device.getRFAddress());
-            // Add additional device config entries
+            if (getConfig().get(PROPERTY_ROOMNAME) == null
```

Comment:
I think the null check is not needed. `equalsIgnoreCase` can handle null arguments and returns `false` if the argument is null.

Reply:
cool, did not know that. I removed the first check